### PR TITLE
added --container-engine

### PR DESCRIPTION
**Pull request by jfennick**

### CODE REVIEW

File: cwl_utils/docker_extract.py [hunk 1]
```diff
@@ -32,6 +32,11 @@ def arg_parser() -> argparse.ArgumentParser:
         help="Use singularity to pull the image",
         action="store_true",
     )
+    parser.add_argument(
+        "--user-space-docker-cmd",
+        dest="user_space_docker_cmd",
```
Comment (mr-c) [inline cwl_utils/docker_extract.py:37]:
```suggestion
        "--container-engine",
```
No reason to mention "user space"

**Review by mr-c — CHANGES_REQUESTED** [review]
Thank you! One small request about the name of the option

File: cwl_utils/docker_extract.py (hunk 1 above)
Comment (jfennick) [reply cwl_utils/docker_extract.py:37]: Gotcha. I was just trying to keep it consistent with [cwltool](https://github.com/common-workflow-language/cwltool/blob/main/cwltool/argparser.py#L433) and [cwltoil](https://github.com/DataBiosphere/toil/blob/master/src/toil/cwl/cwltoil.py#L3179), but I can make the change.

Comment (jfennick) [reply cwl_utils/docker_extract.py]: Done!

**Review by mr-c — APPROVED** [review]

**Comment by jfennick** [comment]
I fixed the formatting, but there appears to be another problem: podman does not support colons in filenames.

`podman save -o quay.iobriandoconnordockstore-tool-md5sum:1.0.4.tar quay.io/briandoconnor/dockstore-tool-md5sum:1.0.4
Error: invalid filename (should not contain ':') "quay.iobriandoconnordockstore-tool-md5sum:1.0.4.tar"`

See https://github.com/containers/podman/issues/489

One solution is to simply replace colons with underscores. For the use case where the goal is to simply pull the images into the local cache, then the filenames in the separate output directory don't necessarily matter.

File: cwl_utils/image_puller.py [hunk 2]
```diff
@@ -57,11 +57,15 @@ def save_docker_image(self) -> None:
         _LOGGER.info(f"Pulling {self.req} with Docker...")
         cmd_pull = [self.cmd, "pull", self.req]
         ImagePuller._run_command_pull(cmd_pull)
+        filename = os.path.join(self.save_directory, self.get_image_name())
+        # Replace colons with underscores in the filename.
```
Comment (mr-c) [inline cwl_utils/image_puller.py:61]: This should go in `get_image_name()`

**Review by mr-c — CHANGES_REQUESTED** [review]

**Review by mr-c — APPROVED** [review]

**Pull request merged**